pkg/omf/functions/packages/omf.packages.list.fish

```diff
@@ -1,69 +1,85 @@
-function __omf.packages.sort
-  for package in (echo $argv | tr ' ' '\n' | sort); echo $package; end
-end
-
-function __omf.packages.builtin
-  echo "omf"
-  echo "fish-spec"
-end
+set -l builtin_package_path {$OMF_PATH,$OMF_CONFIG}/pkg*/{omf,fish-spec}
 
-function __omf.packages.list -a type
-  set -l list
-
-  test "$type" = "--theme"; or for package in (basename -a {$OMF_CONFIG,$OMF_PATH/db}/pkg/*)
-    contains $package (__omf.packages.builtin); or set list $list $package
+function __omf.packages.format
+  set IFS /
+  for path in $argv
+    echo $path | read -la components
+    echo $components[-1]
   end
-
-  test "$type" = "--plugin"; or for package in (basename -a {$OMF_CONFIG,$OMF_PATH/db}/themes/*)
-    set list $list $package
-  end
-
-  __omf.packages.sort $list
 end
 
 function __omf.packages.list.available -a type
   set -l list
+  set -l database_package_path $OMF_PATH/db/pkg/*
+  set -l database_theme_path $OMF_PATH/db/themes/*
+  set -l installed_package_path {$OMF_CONFIG,$OMF_PATH}/pkg/*
+  set -l installed_theme_path {$OMF_CONFIG,$OMF_PATH}/themes/*
 
-  test "$type" = "--theme"; or for package in (basename -a $OMF_PATH/db/pkg/*)
-    contains $package (basename -a {$OMF_CONFIG,$OMF_PATH}/pkg/*); or set list $list $package
+  test "$type" = "--theme"; or for path in $database_package_path
+    contains $path $installed_package_path; or set list $list $path
   end
 
-  test "$type" = "--plugin"; or for package in (basename -a $OMF_PATH/db/themes/*)
-    contains $package (basename -a {$OMF_CONFIG,$OMF_PATH}/themes/*); or set list $list $package
+  test "$type" = "--plugin"; or for path in $database_theme_path
+    contains $path $installed_theme_path; or set list $list $path
```

Comment:
Here and in the previous `test` block, it's comparing two full paths from globbing two different folders. Need to `__omf.packages.format` the lists beforehand.